tools/probes/probes_main.c

```diff
@@ -0,0 +1,288 @@
+{
+	uint32_t received_crc;
+	uint32_t calc_crc;
+
+	received_crc = data_packet->checksum;
+	data_packet->checksum = 0;
+	calc_crc = crc32(0, (char *)data_packet, sizeof(*data_packet));
+
+	if (received_crc == calc_crc)
+		return 1;
+	else
+		return 0;
+}
+
+void parse_data(char *file_in)
+{
+	FILE *fd_in;
+	struct wave_files files[FILES_LIMIT];
+	struct probe_data_packet *packet;
+	uint32_t data[DATA_READ_LIMIT];
+	uint32_t total_data_to_copy = 0;
+	uint32_t data_to_copy = 0;
+	uint32_t *w_ptr;
+
+	int i, j, file;
+
+	enum p_state state = 0;
+
+	fprintf(stdout, "%s:\t Parsing file: %s\n", APP_NAME, file_in);
+
+	fd_in = fopen(file_in, "rb");
+	if (!fd_in) {
+		fprintf(stderr, "error: unable to open file %s, error %d\n",
+			file_in, errno);
+		exit(0);
+	}
+
+	memset(&data, 0, sizeof(uint32_t) * DATA_READ_LIMIT);
+	memset(&files, 0, sizeof(struct wave_files) * FILES_LIMIT);
+	file = -1;
+
+	do {
+		i = fread(&data, sizeof(uint32_t), DATA_READ_LIMIT, fd_in);
+		for (j = 0; j < i; j++) {
+			if (data[j] == PROBE_EXTRACT_SYNC_WORD) {
+				if (state != READY) {
+					fprintf(stderr, "error: wrong state %d, err %d\n",
+						state, errno);
+					free(packet);
+					exit(0);
+				}
+				packet = malloc(sizeof(struct probe_data_packet));
+				memset(packet, 0, sizeof(struct probe_data_packet));
+				total_data_to_copy = sizeof(struct probe_data_packet) /
+					sizeof(uint32_t);
+				w_ptr = (uint32_t *)packet;
+				state = SYNC;
+			}
+
+			if (total_data_to_copy > 0) {
+				if (j + total_data_to_copy > i) {
+					data_to_copy = i - j;
+					total_data_to_copy -= data_to_copy;
+				} else {
+					data_to_copy = total_data_to_copy;
+					total_data_to_copy = 0;
+				}
+				memcpy(w_ptr, data + j, data_to_copy * sizeof(uint32_t));
+				w_ptr += data_to_copy;
+				j += data_to_copy - 1;
+			}
+
+			if (total_data_to_copy == 0) {
+				switch (state) {
+				case READY:
+					break;
+				case SYNC:
+					total_data_to_copy = packet->data_size_bytes /
+						sizeof(uint32_t);
+					packet = realloc(packet,
+							 sizeof(struct probe_data_packet) +
+							 packet->data_size_bytes);
+					w_ptr = (uint32_t *)&packet->data;
+					state = CHECK;
+					break;
+				case CHECK:
+					if (validate_data_packet(packet)) {
+						file = get_buffer_file(files,
+								       packet->buffer_id);
+
+						if (file < 0)
+							file = init_wave(files,
+									 packet->buffer_id,
+									 packet->format);
+
+						fwrite(packet->data,
+						       sizeof(uint32_t),
+						       packet->data_size_bytes /
+						       sizeof(uint32_t),
+						       files[file].fd);
+
+						files[file].size += packet->data_size_bytes /
+							sizeof(uint32_t);
+					}
+					free(packet);
+					state = READY;
+					break;
+				}
+			}
+		}
+	} while (i > 0);
+
+	finalize_wave_files(files);
+
+	fclose(fd_in);
+	fprintf(stdout, "%s:\t done\n", APP_NAME);
+}
+
+int main(int argc, char *argv[])
+{
+	int opt;
+
+	while ((opt = getopt(argc, argv, "d:e:hi:p:")) != -1) {
```

Comment:
It looks that most of these args are unused (just p & h are in use atm).

Reply:
sorry, forgot about this